drivers/wifi/nrf700x/src/wpa_supp_if.c

```diff
@@ -2333,6 +2510,7 @@ int nrf_wifi_wpa_supp_sta_remove(void *if_priv, const u8 *addr)
 	ret = 0;
 
 out:
+	k_mutex_unlock(&vif_ctx_zep->vif_lock);
```

Comment:
Ditto, at line 2490. It seems to be the case for other functions as well, it seems that the argument check needs to be revised (maybe just return early?).